IdleBanning.js

```diff
@@ -148,35 +178,40 @@ var banUserInternally = function (args, behaviorOverride) {
 		&& args.leaderboardName.length > 0
 		&& args.leaderboardName.trim()
 	) {
-		// NOTE : we only reset tier if there is leaderboard specified.
-		// This is due to the client (v35) auto banning not working correctly
-		var playerTier = getPlayerTierIndex();
+		// find player entries in global leaderboard
+		var playerToResetToScore = getPlayersWithScoreToReset(args.leaderboardName);
 
-		var playerRedisKey = getPlayerLeaderboardId();
+		// get max player score in leaderboard
+		// and reset all scores
+		var maxScore = playerToResetToScore[getPlayerLeaderboardId()];
 
-		sendUwsUpdateLeaderboardRequest(
-			playerRedisKey,
-			args.leaderboardName,
-			0,
-			'Last',
-			playerTier
-		);
+		var playerTier = getPlayerTierIndex(true);
 
-		var playerRedisKeys = playerRedisKey.split('|');
-		if (playerRedisKeys.length > 1) {
+		for(var fieldName in playerToResetToScore) {
 			sendUwsUpdateLeaderboardRequest(
-				playerRedisKeys[0],
+				fieldName,
 				args.leaderboardName,
 				0,
 				'Last',
-				playerTier
+				playerTier,
+				true
 			);
-		}
+		};
 
-		updatePlayerTierData(null, {'tier': -1}, "banUser:"+args.leaderboardName);
+		// send write request with player flagged as cheater
+		banData['leaderboard'] = args.leaderboardName;
+		banData['score'] = maxScore;
 
-		banData['tier'] = playerTier;
-		banData['leaderbord'] = args.leaderboardName;
+		sendUwsUpdateLeaderboardRequest(
+			getPlayerLeaderboardId(),
+			args.leaderboardName,
+			maxScore
+		);
+
```

Comment:
Can we add a Loggly Here - would be good to have a way to get banned user

Reply:
updated.  also, each player in admin tool displays ban data